src/widgets/unfold_widgets/import_widget.py

```diff
@@ -185,15 +177,17 @@ def click_listener(self, button_type: str):
 
     def load_from_file_handle(self):
         self.import_state_label.setText("Loading ...")
-        filepath = self.filepath_line.text()
-        result = self.engine.load_data_from_file(filepath)
-        if result:
-            self.import_state_label.setText(result)
-            return
-        self.clear_widgets()
-        self.set_options()
-        self.set_columns_grid()
-        self.display_data()
+        file_path: str = QFileDialog.getOpenFileName(self, 'Choose file', '.', "*.csv *.json")[0]
+        try:
+            self.engine.load_data_from_file(file_path)
+        except ValueError as e:
+            self.import_state_label.setText(str(e))
+        else:
+            self.filepath_line.setText(file_path)
```

Comment:
My suggestion is to show only name of file, because whole filepath is unreadable and isn't fit in field.